targets/TARGET_STM/TARGET_STM32F4/TARGET_STM32F429xI/device/TOOLCHAIN_ARM_STD/stm32f429xx.sct

```diff
@@ -36,6 +36,8 @@
   #define MBED_APP_SIZE 0x200000
 #endif
 
+#if !defined(__MICROLIB)
```

Comment:
I wonder if it would be easier to have two separate scatter files. This means that the arm.py file would need to pick the correct scatter file when building for bare metal. Let's discuss

Reply:
Mbed OS build tools will not have the option to pick a specific set of file from the build directory based on config
In our use case, let's assume stm32429xx.sct and stm32429xx_micor.sct under TOOLCHAIN_ARM_STD but we could not have the option to choose to specific scatter file in the build system as it fetches as a directory and build all the source under the directory.

still, if we can allow build tool to build both stm32429xx.sct and stm32429xx_micor.sct then it creates compile error due to macro and function redefinition.

Reply:
Since this was discussed with tools team and decided that this is not a viable option, I am okay with the current approach.